### ISSUE TITLE
Front End: Unit test coverage for VA Police data

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#16208]
## User Story or Problem Statement
This is a placeholder for 80% unit test coverage requirement for the new VA Police data pages on VAMC facilities.

## Acceptance Criteria
- ~e2e test is completed: When pages are created for systems. Check any system page that side panel element is created and that page exists (this can be done only after skeleton is created and graphQL is changed to pull in side panel link)~ - now tracked in #16303 
- unit tests are completed:
  - CONTENT OF JSON TESTING: Separate function for conversion (postProcess function into named function). Test error/null state. Create Mock CSV data (from CSV @omahane created) and run test on new function to see expected conversion.
  - ~CONTENT OF SELECTOR TESTING: (requires: system-facility data from view, mock data for unit test of selector). Test non-existent facility. Test existing facility and expect mock data.~ - now tracked in #16303

**Comment by maxx1128** [comment]
I would point this to a 3. Unit tests are relatively simple to write and run, but the complexity of the police data raises the points.

**Comment by eselkin** [comment]
I think it would be a 3, but might be more if we uncover more than listed below.

Existing e2e test is:
1. Does the vamc-police.json file get generated and is served from /vamc-police.json

New e2e test:
1. When pages are created for systems. Check any system page that side panel element is created and that page exists (this can be done only after skeleton is created and graphQL is changed to pull in side panel link)

The unit tests will be:
1. CONTENT OF JSON TESTING: Separate function for conversion (postProcess function into named function). Test error/null state. Create Mock CSV data (from CSV @omahane created) and run test on new function to see expected conversion.
2. CONTENT OF SELECTOR TESTING: (requires: system-facility data from view, mock data for unit test of selector). Test non-existent facility. Test existing facility and expect mock data.

**Comment by eselkin** [comment]
I think we need to break this testing up.
I built the unit tests for the processing of CSVs for contacts and police data into JSON and testing the JSON. The problem with the other tests is
e2e is blocked until CMS figures out the VA Police sidebar issue
e2e is blocked until skeleton is created
e2e is blocked until widget is created for VA Police skeleton page
#2 unit test is blocked until widget is created for VA Police Skeleton page.

**Comment by jilladams** [comment]
Went ahead and broke the e2e and Selector test ACs out into a separate ticket, #16303 . Updated ACs in ticket body.

**Comment by xiongjaneg** [comment]
Subsequent tickets were created to capture blocked work. Closing.
